Libraries/VtkVgModelView/vtkVgTrackFilter.cxx

```diff
@@ -123,35 +113,40 @@ int vtkVgTrackFilter::GetBestClassifier(vtkVgTrack* track)
   int bestType = -1;
   double bestScore = -1.0;
 
-  double pvo[3];
-  track->GetPVO(pvo);
+  const auto& toc = track->GetTOC();
 
-  // if unclassified, doesn't matter what he filter setting are
-  if (pvo[0] == 0 && pvo[1] == 0 && pvo[2] == 0)
+  // If unclassified, doesn't matter what the filter setting are
+  if (!toc.size())
     {
-    return vtkVgTrack::Unclassified;
+    return std::numeric_limits<int>::max();
```

Comment:
If we're going to have different sentinel values to denote different error conditions, I think we should make them all negative for consistency (-1, -2, etc., because using `INT_MAX` seems goofy to me), and they should also be labeled (`return NO_TYPE` instead of `return -1`).

Reply:
They are definitely not exceptional (besides that I am — as are Qt and ViViA — not a fan of exceptions); having tracks loaded with no TOC's is common, as is having tracks that don't pass any filters.

I'd be concerned that using other negative numbers is going to break all sorts of code that interprets *less than zero* as "filtered"... there are quite a few places in ViViA and Qt that "just happen" to use `-1` for certain conditions, but that *document* (either implicitly or explicitly) that those conditions are described by "a value less than zero".

The use of `INT_MAX` is meant to follow the existing pattern defined by `vtkVgTrack::Unclassified`, which is "a positive value that does not map to a known type". With PVO's, we were able to use `4` for that purpose. With TOC's, we use `INT_MAX` and are admittedly crossing our fingers that that value will never be used for a regular type.